# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1421618054 -3600
#      Sun Jan 18 22:54:14 2015 +0100
# Branch gui-release
# Node ID 7bb80c927af5543d036bab512a18dd26e12bc3c6
# Parent  f9a944b9e1cbcfcc3408e9cb825355e9aa5d282d
fix cursor positioning when searching backwards (bug #44039)

* find-dialog.cc (find): fix getting the actual cursor position when
  searching backwards

diff --git a/libgui/src/m-editor/find-dialog.cc b/libgui/src/m-editor/find-dialog.cc
--- a/libgui/src/m-editor/find-dialog.cc
+++ b/libgui/src/m-editor/find-dialog.cc
@@ -285,18 +285,19 @@ find_dialog::find (bool forward)
               col  = _edit_area->text (line).length () - 1;
               if (col == -1)
                 col = 0;
             }
         }
       else if (! do_forward)
         {
            // search from previous character if search backward
+           _edit_area->getCursorPosition (&line,&col);
            int currpos = _edit_area->positionFromLineIndex(line,col);
-           if(currpos > 0) currpos --;
+           if (currpos > 0) currpos --;
            _edit_area->lineIndexFromPosition(currpos, &line,&col);
         }
     }
 
   if (_edit_area)
     {
       if (_edit_area->hasSelectedText ()
           && _search_selection_check_box->isChecked ())
